bin/build_cmd.mli

```diff
@@ -1,7 +1,8 @@
 open Import
 
 val run_build_system
-  :  common:Common.t
+  :  pre_build:(unit -> unit Fiber.t)
```

Comment:
Can you remove this argument and accomplish the goal in some other way?

Reply:
The first thing I did, was calling Scheduler.go to resolve the fiber, doing that `dune fmt` command end up calling `Scheduler.go` twice, it works but IMHO it is not appropriate.

Following your comment below one thing came to my mind, maybe by resolving the aliases into `run_build_system`  but `pre_build` will be replaced by `targets`.

Reply:
I did hesitate about this also at some point but don't IMHO it is appropriate.
```ocaml
    let request setup =
      let open Action_builder.O in
      let* () =
        if target_alias_is_fmt targets
        then
          Lock_dev_tool.lock Dune_pkg.Dev_tool.Fmt
          |> Memo.of_non_reproducible_fiber
          |> Action_builder.of_memo
        else Action_builder.return ()
      in
      Target.interpret_targets (Common.root common) config setup targets
    in
    run_build_command ~pre_build ~common ~config ~request
```

Reply:
Yes, calling `Scheduler.go` (or `Memo.run`) is certainly inappropriate. What is more appropriate is to split the functions that you need so you can only get the functionality you need